sage/pallet/src/lib.rs

```diff
@@ -107,43 +593,75 @@ pub mod pallet {
 			)
 			.map_err(|e| Error::<T, I>::Transition { code: e.as_error_code() })?;
 
+			// TODO: Review the logic in this section
+			let current_season_id = T::SeasonHandler::get_current_season_id();
+			let transition_fee = {
+				let SeasonConfigOf::<T, I> { fee, .. } =
+					T::SeasonHandler::get_season_config_for(&current_season_id)?;
+
+				let base_fee = fee.state_transition;
+				let updated_fee = T::FeeHandler::try_propagate_tournament_fee(
+					base_fee,
+					&sender,
+					&current_season_id,
+				)?;
+				T::FeeHandler::try_propagate_chain_fee(
+					updated_fee,
+					&sender,
+					&AffiliateMethods::StateTransition,
+				)?
+			};
+			T::FeeHandler::deposit_fee_into_treasury(&sender, &current_season_id, transition_fee)?;
+
 			Self::deposit_event(Event::TransitionExecuted { account: sender, id: transition_id });
```

Comment:
So there is a few things to consider here, until now, and this indicates that we should be open for change, until now, a season technically never ended in terms of playing it, why? Because the only thing that the season limited was the minting transition, the forging of a season never stopped, and so did the trading of the season avatars.

I think this creates the next question that arise from the transitions side, because transitions in fact have a life cycle that can be limited by the season. Which then should answer the fee question as the fees are bound to the transition.